src/components/account/content/profile-content.tsx

```diff
@@ -0,0 +1,203 @@
+      if (props.handleStatusDisplay) {
+        props.handleStatusDisplay('success', 'Settings saved');
+      }
+      setUser(response.data as User);
+      if (authContext.setMember) {
+        authContext.setMember(updatedUser);
+      }
+    } catch (err) {
+      if (props.handleStatusDisplay) {
+        props.handleStatusDisplay('error', err.message);
+      }
+    }
+  };
+
+  const handleTechnologiesError = (message: string) => {
+    if (props.handleStatusDisplay) {
+      props.handleStatusDisplay('error', message);
+    }
+  };
+
+  const handleUsernamValidation = async (
+    username: string,
+    currentUser: User | undefined,
+  ) => {
+    const api = ServiceResolver.apiResolver();
+    const u: Username = {
+      username,
+    };
+
+    if (username == currentUser?.username) {
+      setUsernameAvailability({ valid: true, reason: '' });
+      return;
+    }
+
+    try {
+      const response = (await api.validateUsername(u)) as ApiResponse<
+        UserValidation
+      >;
+      setUsernameAvailability(response.data as UserValidation);
+    } catch {
+      setUsernameAvailability({
+        valid: false,
+        reason: 'Failed to validate username',
+      });
+    }
+  };
+
+  const debouncedUsernameValidation = useCallback(
+    debounce(handleUsernamValidation, 450),
+    [],
+  );
+
+  const handleUsernameChange = (username: string) => {
+    setUsername(username);
+    setUsernameAvailability({ valid: true, reason: 'checking...' });
+    debouncedUsernameValidation(username, user);
+  };
+
+  const handleSelectChange = (e: ValueType<OptionType>) => {
+    const userId = user?.id === undefined ? '' : user.id;
+    const updatedTechnologies: UserTechnology[] = Array.isArray(e)
+      ? e.map((v) => {
+          const userTech = technologies.find((t) => t.name == v);
+          if (userTech == undefined) {
+            return { name: v, userId };
+          } else {
+            return { name: v, userId, id: userTech.id };
+          }
+        })
+      : [];
+
+    setTechnologies(updatedTechnologies);
+  };
+
+  return (
+    <Form>
+      <Image
+        src={(user && user.profilePictureUrl) || defaultProfileImage}
+        width="64"
+        height="64"
+      />
+      <FormLabel htmlFor="username">Username</FormLabel>
+      <FormInput
+        type="text"
+        name="username"
+        id="username"
+        value={username}
+        onChange={(e) => handleUsernameChange(e.target.value)}
+        hasError={!usernameAvailablity.valid}
+      />
+      {usernameAvailablity.reason != '' && username != user?.username && (
+        <UsernameCheck isValid={usernameAvailablity.valid}>
+          {usernameAvailablity.reason}
+        </UsernameCheck>
+      )}
+      <br />
+      <br />
+
+      <FormLabel htmlFor="bio">Bio</FormLabel>
+
+      <FormTextArea
+        name="bio"
+        id="bio"
+        onChange={(e: ChangeEvent<HTMLInputElement>) => setBio(e.target.value)}
```

Comment:
This suggestion requires an update to the type of `<FormTextArea>`'s props as well. 

```suggestion
        onChange={(e) => setBio(e.target.value)}
```